crates/runtime/src/instance.rs

```diff
@@ -74,6 +75,19 @@ pub(crate) struct Instance {
     /// interrupted.
     pub(crate) interrupts: Arc<VMInterrupts>,
 
+    /// A handle to the (over-approximized) set of `externref`s that Wasm code
+    /// is using.
+    ///
+    /// The `vmctx` also holds a raw pointer to the table and relies on this
+    /// member to keep it alive.
+    pub(crate) externref_activations_table: Rc<VMExternRefActivationsTable>,
```

Comment:
FWIW I've been trying to clean this up over time where `Instance` doesn't do much memory management of its internals. Is it possible to package up this and the stack map registry elsewhere? I'm not really sure entirely what that would look like but I think it would be best if we could keep `Instance` having a pretty "raw" feeling rather than holding onto various items.

(e.g. shoving this into `Store` and passing it through for when we call wasm)

Reply:
Reading over this, I also don't think that this needs to hold onto this information? It looks like the store can hold onto it for all instances?

Reply:
Also, to keep adding to this, I'd like to get around to cleaning up the interrupts part eventually too

Reply:
This runs into the weirdness at the `wasmtime` and `wasmtime-runtime` boundary again: this is inside `wasmtime-runtime` and so it has "no" knowledge of `Store` and whether anything else is holding the table alive for it.

Would you prefer that `InstanceHandle::new` took a `*mut VMExternRefActivationsTable` instead of an `Rc` and we added this to the safetry invariants required to be maintained for `InstanceHandle::new`?